### DISCUSSION TITLE
Sound popping

### DISCUSSION THREAD

**Discussion by GiulianoPastorino** [dosbox-staging/dosbox-staging#3865] [Q&A]
In one particular LucasArts game, namely Star Wars: Rebel Assault, I hear a lot of "pops" or "clicks" during play. I hear them when playing with DOSBox Staging or with the old DOSBox 0.74-3. I always thought it was a bad recording by LucasArts.

But, in the same computer, I've installed VMware Worsktation Pro, and insided it I've installed CentOS 7, where I use smushplay ([https://github.com/clone2727/smushplay](https://github.com/clone2727/smushplay)) to play the *.anm files directly from the CD-ROM, and lo and behold, a clean sound... no pops nor clicks.

Could it be a filtering issue? I've tried both `sb_filter = sb16` and `sb_filter = modern` in `[sblaster]`, without success.

Cheers

**Comment by johnnovak** [comment]
> But, in the same computer, I've installed VMware Worsktation Pro, and insided it I've installed CentOS 7, where I use smushplay (https://github.com/clone2727/smushplay) to play the *.anm files directly from the CD-ROM, and lo and behold, a clean sound... no pops nor clicks.

@GiulianoPastorino 
That doesn't count, different playback code. You're not even using DOS, but some Linux player that can handle the file format...

Can you play the game itself in VMware? Does it click and pop then? 

Then how about PCem and 86box? If you can get the in-game sound to be artifact free in any one these other PC emulators, then please raise an issue ticket.

**Reply by johnnovak** [reply]
Btw, @GiulianoPastorino, if you're using `cycles = max`, stop doing that. Set the lowest fixed cycles setting instead that runs the game at an acceptable speed (e.g., `cycles = 50000`). That should reduce or completely eliminate audio artifacts in many affected games.

**Reply by GiulianoPastorino** [reply]
Hi @johnnovak,

Thank you for your replies.

> Btw, @GiulianoPastorino, if you're using cycles = max, stop doing that.

I'm not. I've been using cycles = 40000. But, to be thorough, yesterday I tried with lower cycles (30000, 20000, etc.), but same clicking/popping.

> Then how about PCem and 86box?

Yesterday I tried 86Box with DOS 6.22, emulating a 486 DX/33. Same clicking/popping. Perhaps it is because this game was recorded in 11025 Hz, as I read somewhere at Vogons.

> Can you play the game itself in VMware? Does it click and pop then?

I haven't tried it yet. I will install DOS 6.22 inside VMware and try it this week, and then inform the results here.

Cheers

**Reply by GiulianoPastorino** [reply]
> I haven't tried it yet. I will install DOS 6.22 inside VMware and try it this week, and then inform the results here.

I've just tried it: VMware Workstation with DOS 6.22. It was really bad!! (laughing) I completely forgot that VMware doesn't emulate a slower CPU, so the game was too fast, and the sound too broken.

I believe we can close this issue. The clicking/popping is, actually, not annoying—since the sound on Linux/smushplay was so clean, I was just wondering if I was making some mistake in Staging's config.

Cheers

**Reply by johnnovak** [reply]
> The clicking/popping is, actually, not annoying—since the sound on Linux/smushplay was so clean, I was just wondering if I was making some mistake in Staging's config.

@GiulianoPastorino 
Yeah, so again, modern players can _present_ the same data in an improved way. That doesn't mean the emulation is broken, just old DOS software did not do the advanced post-processing that modern players tend to do.

Case in point: MPEG video streams in old FMV games. They look really blocky, dithered, etc. compared to simply playing the same file in VLC.

Same deal with MOD files, especially when compared to Sound Blaster MOD players.

UPDATE: Out of curiosity I've just had a brief look at this modern player you're using. It uses a resampler adapted from the SoX resampler for the low sample audio in these files. That must be miles better than what the original DOS player must be doing, so yeah, it will sound better.

https://github.com/clone2727/smushplay/blob/a0de944a3f91c3b823e5f63ab13daa8548a3a6c2/rate.cpp#L24

**Comment by Grounded0** [comment]
It can also be something like cycles max 30-50 %.

On a modern computer that quite accurately maps to:

Pentium 90 MHz: 30 %
Pentium 166 MHz: 40 %
Pentium Pro 200 MHz (executing 32-bit code): 50 %

Running old protected mode games with cycles max on a modern computer will likely cause sound problems.
